packages/protocol/contracts/verifiers/SgxVerifier.sol

```diff
@@ -154,19 +155,35 @@ contract SgxVerifier is EssentialContract, IVerifier {
         // Do not run proof verification to contest an existing proof
         if (_ctx.isContesting) return;
 
+        if (_ctx.length != _tran.length) {
+            revert SGX_INVALID_INPUT();
+        }
+
         // Size is: 89 bytes
-        // 4 bytes + 20 bytes + 65 bytes (signature) = 89
-        if (_proof.data.length != 89) revert SGX_INVALID_PROOF();
+        // 4 bytes + 20 bytes + 20 bytes + 65 bytes (signature) = 109
```

Comment:
Since the current SGX prover doesn't support multiple blocks, can we ensure the length of `_ctx` and `_tran` are both 1 for now?

Reply:
The inputs into the verification are still changed so a bit tricky. If length is 1 we could use the old verification path, but then this would be a special case and proof aggregation on a single proof would not be valid. Seems easier if aggregation is always used even if there's only a single proof, otherwise we need multiple verification paths.